### PR TITLE
10.0 pos debt notebook journal 9

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

File: base_attendance/views/res_attendance_view.xml [hunk 1]
```diff
@@ -214,6 +214,11 @@
         </field>
     </record>
 
+    <!--Partners view form-->
+    <record id="base.action_partner_customer_form" model="ir.actions.act_window">
```
Comment (yelizariev) [inline base_attendance/views/res_attendance_view.xml]: You shall not change built-in action. Make a copy instead

File: pos_debt_notebook/static/src/js/pos.js [hunk 2]
```diff
@@ -189,12 +194,9 @@ odoo.define('pos_debt_notebook.pos', function (require) {
...
             data.updates_debt = this.updates_debt();
-            var sum = _.filter(this.get_paymentlines(), function(pl){
-                return pl.cashregister.journal.credits_via_discount;
-            });
-            data.amount_via_discount = _.reduce(sum, function(memo, o){
-                return memo + o.amount;
-            },0);
+            data.amount_via_discount = this.get_summary_for_discount_credits() > 0
+            ? this.get_summary_for_discount_credits()
+            : 0;
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js]: data.amount_via_discount = Math.max(0, this.get_summary_for_discount_credits())

-- it allows to don't compute twice

File: pos_debt_notebook/static/src/js/pos.js [hunk 3]
```diff
@@ -166,6 +166,11 @@ odoo.define('pos_debt_notebook.pos', function (require) {
                     return line.cashregister.journal.debt;
                 });
         },
+        has_credits_via_discounts: function(){
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js]: ``has`` usually means that method returns true or false

rename to paymentlines_with_credits_via_discounts

File: pos_debt_notebook/static/src/js/pos.js [hunk 4]
```diff
@@ -721,6 +731,14 @@ odoo.define('pos_debt_notebook.pos', function (require) {
             this.next_button_html = this.render_validation_button();
             this.$el.append(this.next_button_html);
         },
+        render_change: function() {
+            // deduct the number of discount credits, because they were spent on discounts, but still presence like unspent
+            var order = this.pos.get_order();
+            var change = order.get_summary_for_discount_credits() > 0
+            ? order.get_change() - order.get_summary_for_discount_credits()
+            : order.get_change();
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/js/pos.js]: var change = order.get_change() - Math.max(order.get_summary_for_discount_credits(), 0)

File: pos_debt_notebook/static/src/xml/pos.xml [hunk 5]
```diff
@@ -212,6 +212,48 @@
...
+                              <t t-esc="widget.format_currency(line.get_amount())"/>
+                          </td>
+                      </tr>
+                    </t>
+                </t>
+            </table>
+        </t>
+        <t t-jquery="table.receipt-change" t-operation="replace">
+            <t t-set="credits_via_discounts_presence" t-value="order.has_credits_via_discounts()"/>
+            <t t-set="credits_via_discounts_amount" t-value="order.get_summary_for_discount_credits() &lt; 0 ? 0 : order.get_summary_for_discount_credits()"/>
```
Comment (yelizariev) [inline pos_debt_notebook/static/src/xml/pos.xml]: Try to don't call get_summary_for_discount_credits twice

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Comment by yelizariev** [comment]
Receipt seems ok for now
reward-receipt

File: base_attendance/views/res_attendance_view.xml [hunk 6]
```diff
@@ -222,7 +222,7 @@
 
     <menuitem id="menu_hr_attendance_view_attendances" name="Attendances" parent="menu_hr_attendance_manage_attendances" sequence="10" groups="base_attendance.group_res_attendance" action="hr_attendance_action"/>
 
-    <menuitem id="menu_hr_attendance_view_partners_kanban" name="Partners" parent="menu_hr_attendance_manage_attendances" sequence="15" groups="base_attendance.group_hr_attendance_user" action="base.action_partner_customer_form"/>
+    <menuitem id="menu_hr_attendance_view_partners_kanban" name="Partners" parent="menu_hr_attendance_manage_attendances" sequence="15" groups="base_attendance.group_hr_attendance_user" action="res_partner_action_kanban_view"/>
```
Comment (yelizariev) [inline base_attendance/views/res_attendance_view.xml]: Weird

**Review by yelizariev — COMMENTED** [review]
Code seems ok

**Review by Railia — APPROVED** [review]

**Pull request merged**